client/post-editor/editor-drawer/featured-image.jsx

```diff
@@ -32,25 +36,50 @@ class EditorDrawerFeaturedImage extends Component {
 		stats.recordEvent( 'Featured image removed' );
 	}
 
+	onFilesDrop = ( files ) => {
+		Dispatcher.register( function( payload ) {
+			const action = payload.action;
+
+			switch ( action.type ) {
+				case 'CREATE_MEDIA_ITEM':
+					// called when the transient blob has been created and the upload starts
+					break;
+				case 'RECEIVE_MEDIA_ITEM':
+					// called when the media file has been uploaded and needs to be refreshed from the server
+					setTimeout( () => {
+						PostActions.edit( {
+							featured_image: action.data.ID
+						} );
+					}, 500 );
+			}
+		} );
+
+		MediaActions.clearValidationErrors( this.props.site.ID );
+		MediaActions.add( this.props.site.ID, files ).then();
```

Comment:
I will have to dig a bit further in `WebAudio` for that request 😋  
It's a leftover from researching how `MediaActions` behaved for adding media to a site.